arlo-client/src/components/AuditFlow/index_members.test.tsx

```diff
@@ -40,33 +40,45 @@ describe('AuditFlow initial load', () => {
     apiMock.mockImplementation(memberingMock)
   })
 
-  it('renders correctly', () => {
-    const { container } = render(
-      <Router {...routeProps}>
-        <AuditFlow {...routeProps} />
-      </Router>
-    )
+  it('renders correctly', async () => {
+    let utils: RenderResult
+    await act(async () => {
+      utils = render(
+        <Router {...routeProps}>
+          <AuditFlow {...routeProps} />
+        </Router>
+      )
+    })
+    const { container } = utils!
     expect(container).toMatchSnapshot()
   })
 
   it('fetches initial state from api', async () => {
-    const { container } = render(
-      <Router {...routeProps}>
-        <AuditFlow {...routeProps} />
-      </Router>
-    )
+    let utils: RenderResult
+    await act(async () => {
+      utils = render(
+        <Router {...routeProps}>
+          <AuditFlow {...routeProps} />
+        </Router>
+      )
+    })
+    const { container } = utils!
     await wait(() => {
       expect(apiMock).toBeCalled()
       expect(container).toMatchSnapshot()
     })
   })
 
   it('renders member form', async () => {
-    const { container, getByText } = render(
-      <Router {...routeProps}>
-        <AuditFlow {...routeProps} />
-      </Router>
-    )
+    let utils: RenderResult
+    await act(async () => {
+      utils = render(
+        <Router {...routeProps}>
+          <AuditFlow {...routeProps} />
+        </Router>
+      )
+    })
+    const { container, getByText } = utils!
```

Comment:
Worth wrapping all this up in a helper?

```ts
const { container, getByText } = await renderAtomic(<Router>…</Router>)
```

Could be defined something like this:

```ts
async function renderAtomic(component: React.Component): Promise<RenderResult> {
  let result: RenderResult

  // You have this callback as `async`, but I don't think it needs to be.
  await act(() => {
    result = render(component)
  })

  return result!
}
```

Reply:
@eventualbuddha Something like that could work. [It does actually have to be `async`](https://reactjs.org/docs/testing-recipes.html#act) according to the new pattern, though.

Reply:
My example helper is async. Did you mean "does not actually have to be `async`"? Anyway, this was just a suggestion, and I do tend to have a higher tolerance for repeated code in tests so I think it's fine as-is.